### PR TITLE
Added dn comparison via dn portion. Instead of comparing DN with whit…

### PR DESCRIPTION
**Pull request by fcarbon**
Added dn comparison via dn portion. Instead of comparing DN with whitelist DN as String to String, the DN are compared by portions. In that way, the order of the DN portions, and spaces are no longer relevant when validating a DN.

### CODE REVIEW

File: impersonation/whitelist.go [hunk 1]
```diff
@@ -26,6 +28,51 @@ func CanImpersonate(caller Caller, whitelist Whitelist) bool {
...
+		}
+	}
+
+	if countDnPortions == len(splitWs) {
+		return true
+	}
+	return false
+}
+
+func removeQuirks(dn string) string {
```
Comment (lucasmoten) [inline impersonation/whitelist.go]: I'm not a fan of the way this is written. If handling the parts, delimited by commas, then trim the whitespace around those commas.  And if handling the +, there's got to be a better way then changing to comma which could disrupt other parsing

File: impersonation/whitelist_test.go [hunk 2]
```diff
@@ -107,6 +147,15 @@ func BenchmarkValidateCaller(b *testing.B) {
 	}
 }
 
+
+func getAltServers() []string {
+	return []string{
+		"uid=server1, ou=Server, dc=example, dc=com",
+		"uid=server8+ou=Server+dc=example,dc=com",
```
Comment (lucasmoten) [inline impersonation/whitelist_test.go]: not sure whats going on in this sample

File: impersonation/whitelist.go [hunk 3]
```diff
@@ -26,6 +28,51 @@ func CanImpersonate(caller Caller, whitelist Whitelist) bool {
...
+	dn = strings.Replace(dn, "+", ",", -1)
+	dn = strings.Replace(dn, ", ", ",", -1)
+
+	return dn
+}
+
+func findPortionDn(dnArray [] string, dnPortion string) bool {
+	sort.Strings(dnArray)
+	i := sort.Search(len(dnArray), func(i int) bool { return dnArray[i] >= dnPortion })
+	if i < len(dnArray) && dnArray[i] == dnPortion {
```
Comment (lucasmoten) [inline impersonation/whitelist.go]: This should probably normalize the case, and whitespacing.

Comment (lucasmoten) [inline impersonation/whitelist.go]: Why is this comparing across like this? Is it to try to deal with differing order? Order is important.

What happens if whitelist allows `cn=something,ou=ABC,ou=DEF,c=us`
And checking `cn=something,ou=ABC,c=us`

**Comment by lucasmoten** [comment]
What happened to the working whitelist checks that weve implemented in the dozens of projects before this?

**Comment by alecholmez** [comment]
I'm really hesitant to approve this. DN ordering matters because there are different specs for each case.

File: impersonation/whitelist.go (hunk 3 above)
Comment (joshua-rutherford) [reply impersonation/whitelist.go]: In the JVM code we also allow for reversed ordering.   I presume for some obscure reason I don't recall now.

**Pull request closed without merging**